third_party/tango/libtango_markers/tango_markers.h

```diff
@@ -0,0 +1,134 @@
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef TANGO_MARKERS_TANGO_MARKERS_H_
+#define TANGO_MARKERS_TANGO_MARKERS_H_
+
+#include <tango_client_api.h>
+#include <tango_support.h>
+
+/// @file tango_markers.h
+/// @brief File containing marker detection functions of Tango Support C API.
+///   The Tango Support C API provides helper functions useful to external
+///   developers for building Tango applications. This API is experimental and
+///   subject to change.
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/// @defgroup MarkerDetectionSupport Marker Detection Support Functions
+/// @brief Functions for detecting markers.
+/// @{
+/// @brief A type to define all combinations of markers supported.
+typedef enum {
+  TANGO_MARKERS_MARKER_ARTAG = 0x01,
+  TANGO_MARKERS_MARKER_QRCODE = 0x02,
+} TangoMarkers_MarkerType;
+
+/// @brief A structure to define parameters for passing marker detection
+/// parameters.
+typedef struct TangoMarkers_DetectParam {
+  /// Type of marker to be detected.
+  TangoMarkers_MarkerType type;
+
+  /// The physical size of the marker in meters.
+  double marker_size;
+} TangoMarkers_DetectParam;
+
+/// @brief A structure to define contents of a marker, which can be any of the
+/// marker types supported.
+typedef struct TangoMarkers_Marker {
+  /// The type of the marker.
+  TangoMarkers_MarkerType type;
+
+  /// The timestamp of the image from which the marker was detected.
+  double timestamp;
+
+  /// The content of the marker. For AR tags, this is the string format of the
+  /// tag id. For QR codes, this is the string content of the code.
+  char* content;
+
+  /// The size of content, in bytes.
+  int content_size;
+
+  /// Marker corners in input image pixel coordinates.
+  /// For all marker types, the first corner is the lower left corner, the
+  /// second corner is the lower right corner, the third corner is the upper
+  /// right corner, and the last corner is the upper left corner.
+  ///
+  /// P3 -- P2
+  /// |     |
+  /// P0 -- P1
+  ///
+  float corners_2d[4][2];
+
+  /// Marker corners in the output frame, which is defined by the translation
+  /// and orientation pair passed to TangoMarkers_detectMarkers() function. The
+  /// location of the corner is the same as in corners_2d field.
+  float corners_3d[4][3];
+
+  /// Marker pose - orientation is a Hamilton quaternion specified as
+  /// (x, y, z, w). Both translation and orientation are defined in the output
+  /// frame, which is defined by the translation and orientation pair passed to
+  /// TangoMarkers_detectMarkers() function.
+  /// The marker pose defines a marker local frame, in which:
+  ///  X = to the right on the tag
+  ///  Y = to the up on the tag
```

Comment:
this isn't worded correctly. I would change both of these comments for X and Y.  Maybe just say the tag's up vector and the tag's right vector?

Reply:
This is not our code but the marker detection library header.